src/actions/tamanu/prepare_upgrade.rs

```diff
@@ -0,0 +1,160 @@
+use std::{
+	collections::HashSet,
+	fs,
+	path::{Path, PathBuf},
+};
+
+use clap::Parser;
+use itertools::Itertools;
+use miette::{bail, IntoDiagnostic, Result, WrapErr};
+use node_semver::Version;
+use regex::Regex;
+use tracing::info;
+
+use crate::actions::{
+	tamanu::{
+		download::{download, make_url},
+		find_package,
+	},
+	Context,
+};
+
+use super::{download::ServerKind, find_existing_version, find_tamanu, TamanuArgs};
+
+/// Setup new Tamanu instllation without taking down the existing one
```

Comment:
```suggestion
/// Perform pre-upgrade tasks.
///
/// This will not incur downtime.
///
/// This command will detect which server is installed (Facility or Central) and which version is
/// currently running, then download the desired newer version, unpack it, copy config across,
/// install dependencies, and perform readiness checks.
```